### PR TITLE
Remove no-\r assertion in print_formatted_text

### PR DESCRIPTION
**Pull request by jnwatson**
Follow up from https://github.com/prompt-toolkit/python-prompt-toolkit/issues/915

I'll go ahead and submit a similar PR for 2.0 branch once this one's accepted.

### CODE REVIEW

File: prompt_toolkit/renderer.py [hunk 1]
```diff
@@ -684,9 +684,11 @@ def print_formatted_text(
         else:
             output.reset_attributes()
 
+        # Eliminate newlines
+        text = text.replace('\r', '')
```
Comment (bjorn3) [inline prompt_toolkit/renderer.py]: This would incorrectly remove bare `\r` (eg old mac newline)

**Comment by jnwatson** [comment]
As far as I know, prompt-toolkit doesn't run on pre-OS-X macs. The latest
Python distribution I could find for such a platform was 2.3.3.

File: prompt_toolkit/renderer.py [hunk 2]
```diff
@@ -684,9 +684,11 @@ def print_formatted_text(
         else:
             output.reset_attributes()
 
+        # Eliminate newlines
```
Comment (jonathanslenders) [inline prompt_toolkit/renderer.py]: Technically, these are actually the "carriage returns" which we are eliminating. Would you mind rephrasing the comment? Then it's perfect.

Comment (jnwatson) [reply prompt_toolkit/renderer.py]: ok

**Comment by jonathanslenders** [comment]
Thanks!

**Comment by jonathanslenders** [comment]
And thanks for the unit test too! :-)

**Comment by jonathanslenders** [comment]
I just cherry-picked this commit on the 2.0 branch too, so no need to create a second PR.

**Comment by jnwatson** [comment]
My pleasure. Thank you for all of your hard work.

**Pull request merged**